lambdas/file_access_patterns/lambda_for_fileUpload/index.js

```diff
@@ -31,31 +49,31 @@ exports.handler = async (event) => {
     f_type: reqBody.f_type,
   };
 
-  try{
-    await dynamo.put({
-      TableName:"V-Transfer",
-      Item: file
-    }).promise()
+  let generateUrlTask;
+  const params = { FunctionName: 'lambda_for_generateURL', InvocationType: 'RequestResponse',
+    Payload: getGenerateUrlPayload(file.hash, file.SK.split('#')[1])
   }
-  catch(err){
-    return response(500,"Internal Server Error",undefined);
+  try {
+    const res = await Lambda.invoke(params).promise()
+    generateUrlTask = JSON.parse(res.Payload)
+  } catch(err) { return response(500, JSON.stringify(err), undefined) }
+  file.default = generateUrlTask.Item.GS1_PK
+  const filePutTask = { TableName:"V-Transfer", Item: file }
+  const userMetaDataUpdateTask = { TableName: "V-Transfer",
+    Key: { PK: `USER#${event.path.userId}`, SK: `METADATA` },
+    UpdateExpression: "add storage_used :file_size",
+    ExpressionAttributeValues: { ":file_size": reqBody.size }
   }
-
-  try{
-    await dynamo.update({
-      TableName: "V-Transfer",
-      Key: {
-        PK: `USER#${event.path.userId}`,
-        SK: `METADATA`,
-      },
-      UpdateExpression: "add storage_used :file_size",
-      ExpressionAttributeValues: {
-        ":file_size": reqBody.size,
-      },
-    }).promise()
-    return  response(201, undefined,file)
+  
+  const param = {
+    TransactItems: [{ Put: filePutTask }, { Update: userMetaDataUpdateTask }, { Put: generateUrlTask }]
   }
-  catch(err){
-    return response(500,"Internal Server Error",undefined);
+
+  try {
+    await dynamo.transactWrite(param).promise()
+    return response(200, undefined, file)
+  } catch (err) {
+    return response(500, JSON.stringify(err), undefined)
```

Comment:
Return `Internal Server Error` as the error message.

Reply:
this also